sklearn/_isotonic.pyx

```diff
@@ -62,9 +62,9 @@ def _inplace_contiguous_isotonic_regression(floating[::1] y, floating[::1] w):
             i = k
 
 
-def _make_unique(cnp.ndarray[dtype=floating] X,
-                 cnp.ndarray[dtype=floating] y,
-                 cnp.ndarray[dtype=floating] sample_weights):
+def _make_unique(floating[::1] X,
+                 floating[::1] y,
+                 floating[::1] sample_weights):
```

Comment:
From looking at the code, I do not think the input: `X`, `y`, and `sample_weights` are changed. They can be made `const`.